cmd/terramate/e2etests/exp_clone_test.go

```diff
@@ -93,3 +96,166 @@ generate_hcl "test2.hcl" {
 	test.AssertGenCodeEquals(t, genHCL, `a = "literal"`)
 	test.AssertGenCodeEquals(t, genHCL2, `b = null`)
 }
+
+func TestCloneStacksWithChildren(t *testing.T) {
+	t.Parallel()
+
+	type testcase struct {
+		Name            string
+		Stacks          []string
+		Files           []string
+		Srcdir          string
+		Destdir         string
+		SkipChildStacks bool
+		WantStacks      []string
+		WantFiles       []string
+		WantError       string
+	}
+
+	tests := []testcase{
+		{
+			Name: "clone stack dir with children",
+			Stacks: []string{
+				"stack-1",
+				"stack-1/child-a",
+				"stack-1/child-b",
+				"stack-2",
+				"stack-2/child-c",
+			},
+			Files: []string{
+				"f:stack-1/somefile.txt:foo",
+				"f:stack-1/.ignored",
+			},
+			Srcdir:  "stack-1",
+			Destdir: "cloned-stack-1",
+			WantStacks: []string{
+				"cloned-stack-1",
+				"cloned-stack-1/child-a",
+				"cloned-stack-1/child-b",
+			},
+			WantFiles: []string{
+				"f:cloned-stack-1/somefile.txt:foo",
+				"!e:cloned-stack-1/.ignored",
+			},
+		},
+		{
+			Name: "clone non-stack dir with children",
+			Stacks: []string{
+				"stacks/stack-1",
+				"stacks/stack-2",
+				"stacks/stack-3",
+			},
+			Srcdir:  "stacks",
+			Destdir: "cloned-stacks",
+			WantStacks: []string{
+				"cloned-stacks/stack-1",
+				"cloned-stacks/stack-2",
+				"cloned-stacks/stack-3",
+			},
+		},
+		{
+			Name: "clone stack dir without children",
+			Stacks: []string{
+				"stack-1",
+				"stack-1/child-a",
+				"stack-2",
+				"stack-2/child-b",
+				"stack-2/child-c",
+			},
+			Files: []string{
+				"f:stack-2/somefile.txt:foo",
+				"f:stack-2/.ignored",
+				"f:stack-2/mydir/others.txt:foo",
+				"f:stack-2/mydir/stuff/.ignored2",
+			},
+			Srcdir:          "stack-2",
+			Destdir:         "cloned-stack-2",
+			SkipChildStacks: true,
+			WantStacks: []string{
+				"cloned-stack-2",
+			},
+			WantFiles: []string{
+				"!e:cloned-stack-2/.ignored",
+				"!e:cloned-stack-2/mydir/stuff/.ignored2",
+				"f:cloned-stack-2/somefile.txt:foo",
+				"f:cloned-stack-2/mydir/others.txt:foo",
+				"f:cloned-stack-2/mydir/others.txt:foo",
+			},
+		},
+		{
+			Name: "fail to clone non-stack dir if skipping children",
+			Stacks: []string{
+				"stacks/stack-1",
+				"stacks/stack-2",
+				"stacks/stack-3",
+			},
+			Files: []string{
+				"f:stacks/somefile.txt:foo",
+				"f:stacks/other/other.txt:bar",
+			},
+			Srcdir:          "stacks",
+			Destdir:         "cloned-stacks",
+			SkipChildStacks: true,
+			WantError:       "no stacks to clone",
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+
+		t.Run(tc.Name, func(t *testing.T) {
+			layout := []string{}
+			for _, st := range tc.Stacks {
+				layout = append(layout, fmt.Sprintf("s:%s", st))
+			}
+			layout = append(layout, tc.Files...)
+
+			s := sandbox.New(t)
```

Comment:
Normally, yes, since it doesn't require git. However, since I didn't add a terramate config the root, terramate will fail to run:
> FTL Project root not found. If you invoke Terramate inside a Git repository, Terramate will automatically assume the top level of your repository as the project root. If you use Terramate in a directory that isn't a Git repository, you must configure the project root by creating a terramate.tm.hcl configuration in the directory you wish to be the top-level of your Terramate project.```

I would leave it like this for now, but in general it would be nice to have a non-Git sandbox that works without having to manually add a config file to each testcase.

Reply:
The `sandbox.NoGit(t, true)`, creates the root config.

Reply:
Ah, I see it now - hadn't rebased to latest main yet at the time of the comment.